test/cypress/integration/components/filter-blog-posts.js

```diff
@@ -0,0 +1,302 @@
+    filter.clickTopic( 'Mortgages' );
+    // Then the list of topics should show only tags that contain the word "mortgage"
+    page.results().should( 'contain', 'mortgage' );
+    // And when I select a topic in the list
+    filter.clickTopic( 'Servicemembers' );
+    // And I click "Apply filters" button
+    page.applyFilters();
+    // Then I should see only results tagged with the selected topic
+    page.notification().should( 'be.visible' );
+    page.results().should( 'contain', 'service' );
+    // And the page url should contain "topics=mortgages"
+    cy.url().should( 'include', 'topics=mortgages' );
+    // And the page url should contain "topics=servicemembers"
+    cy.url().should( 'include', 'topics=servicemembers' );
+  } );
+  it( 'Select category and topic', () => {
+    // When I select a checkbox in the Category list
+    filter.checkCategoryId( 'policy_compliance' );
+    // When I select a checkbox in the Topic list
+    filter.clickTopic( 'Students' );
+    // And I click "Apply filters" button
+    page.applyFilters();
+    // Then I should see only results that are both in the selected category and tagged with the selected topic
+    page.notification().should( 'be.visible' );
+    page.results().should( 'contain', 'student' );
+    page.results().should( 'contain', 'policy' );
+    // And the page url should contain "categories=policy_compliance"
+    cy.url().should( 'include', 'categories=policy_compliance' );
+    // And the page url should contain "topics=students"
+    cy.url().should( 'include', 'topics=students' );
+  } );
+  it( 'Clear and hide filters', () => {
+    // When I select the last checkbox in the Category list
+    filter.checkCategoryName( 'Info for consumers' );
+    // When I select a checkbox in the Topic list
+    filter.clickTopic( 'Consumer complaints' );
+    // And I click "Apply filters" button
+    page.applyFilters();
+    // Then the page url should contain "categories=info-for-consumers"
+    cy.url().should( 'include', 'categories=info-for-consumers' );
+    // And the page url should contain "topics=consumer-complaints"
+    cy.url().should( 'include', 'topics=consumer-complaints' );
+    // Then I should see only results that are both in the selected category and tagged with the selected topic
+    page.results().should( 'contain', 'consumer' );
+    // And when I click "Show filters"
+    page.showFilters();
+    // And when I click "Clear filters"
+    page.clearFilters();
+    // And I click "Hide filters" button
+    page.hideFilters();
+    // Then the page url should not contain "categories=info-for-consumers"
+    cy.url().should( 'not.include', 'categories=info-for-consumers' );
+    // And the page url should not contain "topics=consumer-complaints"
+    cy.url().should( 'not.include', 'topics=consumer-complaints' );
+    // Then I should see the full list of results
+    page.notification().should( 'be.visible' );
+  } );
```

Comment:
This test is failing for me when I run it headless (but passing when I run it with the GUI in Firefox), but I wonder if it's due to some of the showing/hiding redundancies I commented on earlier.

Reply:
This test should now pass running headless. I have been testing using Chrome. I did not test using Firefox.

Reply:
This test should now pass running headless as I separated Clear and Hide filters into separate tests.
Show filters and Hide filters are in separate tests.
I have been testing using Chrome. I did not test using Firefox.